arangod/Replication2/StateMachines/Document/DocumentFollowerState.cpp

```diff
@@ -96,4 +106,38 @@ auto DocumentFollowerState::applyEntries(
       });
 }
 
+/**
+ * @brief Using the underlying transaction handler to apply a local transaction,
+ * for this follower only.
+ */
+auto DocumentFollowerState::forceLocalTransaction(OperationType opType,
+                                                  velocypack::SharedSlice slice)
+    -> Result {
+  auto doc = DocumentLogEntry{std::string(shardId), opType, std::move(slice),
+                              TransactionId{0}.asFollowerTransactionId()};
+  if (auto applyRes = _transactionHandler->applyEntry(doc); applyRes.fail()) {
+    _transactionHandler->removeTransaction(
+        TransactionId{0}.asFollowerTransactionId());
+    return applyRes;
+  }
+  auto commit = DocumentLogEntry{std::string(shardId),
+                                 OperationType::kCommit,
+                                 {},
+                                 TransactionId{0}.asFollowerTransactionId()};
+  return _transactionHandler->applyEntry(commit);
```

Comment:
We should use `TransactionId::createFollower()` to create a new follower transactionId and use that one throughout the function.

Reply:
I didn't use that one because I was not sure how the follower transaction is generated - randomly, or incrementally. I want to avoid a scenario when the follower generates such an id and then a transaction coming from the leader uses the same.

Reply:
The transaction id is generated using a hybrid logical clock, so this is safe and there won't be any conflicts.

Reply:
Good to know, will do the change then.